bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -338,6 +340,23 @@ private boolean performSearch(boolean updateFromIncrementalBaseLocation) {
 		return somethingFound;
 	}
 
+	/**
+	 * Restores the original caret/selection position when the search field becomes
+	 * empty.
+	 */
+	private void restoreSelectionIfEmpty() {
+	    if (!isAvailableAndActive(SearchOptions.INCREMENTAL)) {
+			// Incremental mode not active, so skipping restore
+	        return;
+	    }
```

Comment:
This is obsolete as the caller already does the check. Either of the two checks may be removed.